protos/flyteidl/core/identifier.proto

```diff
@@ -10,6 +10,7 @@ enum ResourceType {
   TASK               = 1;
   WORKFLOW           = 2;
   LAUNCH_PLAN        = 3;
+  DATASET            = 4;
```

Comment:
One of these things is not like the other.  Can we at least add a comment explaining why there is something that looks different here?  Also, could we create a GH issue if there isn't one to merge all the proto objects in data catalog https://github.com/lyft/datacatalog/blob/master/protos/idl/service.proto into this repo?

Reply:
Agreed